test/e2e/infraprovider/providers/kind/kind.go

```diff
@@ -119,6 +119,68 @@ func (k *kind) GetK8HostPort() uint16 {
 	return k.hostPort.Allocate()
 }
 
+// getContainerState returns the state of a container by name
+// Returns empty string if container doesn't exist
+func getContainerState(containerName string) (string, error) {
+	stdOut, err := exec.Command(containerengine.Get().String(), "ps", "-a", "-f", fmt.Sprintf("name=^%s$", containerName), "--format", "{{.State}}").CombinedOutput()
```

Comment:
not great; ideally we'd switch everything to use native libraries to talk to docker / podman. I will take a look at it at some point later not to increase the scope of the PR even further.